vm/devices/net/gdma/src/bnic.rs

```diff
@@ -167,27 +194,37 @@ impl InspectMut for BasicNic {
 struct Vport {
     mac_address: MacAddress,
     endpoint: Box<dyn Endpoint>,
+    tasks: Vec<VportTask>,
+    serial_no: u32,
+}
+
+struct VportTask {
     task: TaskControl<TxRxState, TxRxTask>,
     queue_cfg: QueueCfg,
-    serial_no: u32,
 }
 
 impl InspectMut for Vport {
     fn inspect_mut(&mut self, req: inspect::Request<'_>) {
-        req.respond()
+        let mut response = req.respond();
+        response
             .field("mac_address", self.mac_address)
-            .field_mut("endpoint", self.endpoint.as_mut())
-            .field("tx_wq", self.queue_cfg.tx.map(|(wq, _cq)| wq))
-            .field("tx_cq", self.queue_cfg.tx.map(|(_wq, cq)| cq))
-            .field("rx_wq", self.queue_cfg.tx.map(|(wq, _cq)| wq))
-            .field("rx_cq", self.queue_cfg.tx.map(|(_wq, cq)| cq))
-            .merge(&mut self.task);
+            .field_mut("endpoint", self.endpoint.as_mut());
+
+        for task in self.tasks.iter_mut() {
+            response
+                .field("tx_wq", task.queue_cfg.tx.map(|(wq, _cq, _wq_obj)| wq))
+                .field("tx_cq", task.queue_cfg.tx.map(|(_wq, cq, _wq_obj)| cq))
+                .field("rx_wq", task.queue_cfg.rx.map(|(wq, _cq, _wq_obj)| wq))
+                .field("rx_cq", task.queue_cfg.rx.map(|(_wq, cq, _wq_obj)| cq));
+            response.merge(&mut task.task);
+        }
     }
 }
 
 struct QueueCfg {
-    tx: Option<(u32, u32)>,
-    rx: Option<(u32, u32)>,
+    /// (wq_id, cq_id, wq_obj_handle)
+    tx: Option<(u32, u32, u64)>,
```

Comment:
The BNIC emulator currently has conjoined the Tx and Rx queues with a given task, while I believe the Tx and Rx queue state is independently managed. With multi-queue, i think this presents some challenges with lifetime management of these queues.

@jstarks - Should we split the Rx and Tx state and have separate Tasks for each of them?